Bottom: 5b906a4b3c5d76476269bc75e4eb1832ce0dfb7b
Top:    30f03ac7c8c987605ab5f609d26656605cb27a50
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-07 19:17:38 +0000

Workaround crash where contour is passed as NULL to poly_PreContour











---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
